stan/math/prim/mat/prob/normal_id_glm_lpdf.hpp

```diff
@@ -0,0 +1,162 @@
+#ifndef STAN_MATH_PRIM_MAT_PROB_NORMAL_ID_GLM_LPDF_HPP
+#define STAN_MATH_PRIM_MAT_PROB_NORMAL_ID_GLM_LPDF_HPP
+
+#include <stan/math/prim/scal/meta/is_constant_struct.hpp>
+#include <stan/math/prim/scal/meta/partials_return_type.hpp>
+#include <stan/math/prim/scal/meta/operands_and_partials.hpp>
+#include <stan/math/prim/scal/err/check_consistent_sizes.hpp>
+#include <stan/math/prim/scal/err/check_finite.hpp>
+#include <stan/math/prim/scal/fun/constants.hpp>
+#include <stan/math/prim/scal/fun/value_of.hpp>
+#include <stan/math/prim/mat/fun/value_of.hpp>
+#include <stan/math/prim/scal/meta/include_summand.hpp>
+#include <stan/math/prim/scal/meta/scalar_seq_view.hpp>
+#include <cmath>
+#include <string>
+
+namespace stan {
+  namespace math {
+
+    /**
+     * Returns the log PDF of the Generalized Linear Model (GLM)
+     * with Normal distribution and id link function.
+     * If containers are supplied, returns the log sum of the probabilities.
+     * @tparam T_n type of vector of dependent variables (labels);
+     * this can also be a single value;
+     * @tparam T_x type of the matrix of independent variables (features); this
+     * should be an Eigen::Matrix type whose number of rows should match the
+     * length of n and whose number of columns should match the length of beta
+     * @tparam T_beta type of the weight vector;
+     * this can also be a single value;
+     * @tparam T_alpha type of the intercept;
+     * this has to be a single value;
+     * @tparam T_scale type of the scale vector.
+     * @param n vector parameter
+     * @param x design matrix
+     * @param beta weight vector
+     * @param alpha intercept (in log odds)
+     * @param sigma (Sequence of) scale parameters for the normal
+     * distribution.
+     * @return log probability or log sum of probabilities
+     * @throw std::domain_error if x, beta or alpha is infinite.
+     * @throw std::domain_error if the scale is not positive.
+     * @throw std::invalid_argument if container sizes mismatch.
+     */
+    template <bool propto, typename T_n, typename T_x, typename T_beta,
+              typename T_alpha, typename T_scale>
+    typename return_type<T_n, T_x, T_beta, T_alpha, T_scale>::type
+    normal_id_glm_lpdf(const T_n &n, const T_x &x, const T_beta &beta,
+                             const T_alpha &alpha, const T_scale& sigma) {
+      static const std::string function = "normal_id_glm_lpdf";
```

Comment:
I just realized I was a jerk and merged my PR before merging this one and now I need to ask you if you can change your `const std::string`s to `const char*`. I'd do it but it's on your fork so I'd have to make a PR, probably easier to just get you to do it if you don't mind. Sorry about that. This will be true in other PRs as well; we reverted #584 because it was causing a huge slow-down.

Reply:
No problem at all. I'll change it now.

Reply:
Hi @seantalts , I've made the requested change and believe the PR can now be merged.